pkg/plugins/secrets/k8s/store.go

```diff
@@ -91,9 +92,9 @@ func (s *KubernetesStore) Update(ctx context.Context, r core_model.Resource, fs
 		return errors.Wrap(err, "failed to convert core Secret into k8s counterpart")
 	}
 	secret.Namespace = s.namespace
-	if r.Descriptor().Scope == core_model.ScopeMesh {
-		setMesh(secret, r.GetMeta().GetMesh(), opts.Labels)
-	}
+
+	setLabelsAnnotationsAndMesh(secret, r.GetMeta().GetMesh(), opts.Labels)
```

Comment:
behavior is weird here. previously we overwrote with `opts.Labels` but only when it's a mesh scope. now we always overwrite with `opts.Labels`. I think https://github.com/kumahq/kuma/pull/10335 should be backported ultimately.